### PR TITLE
[server] Make headless log not yet available 444 status code

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Follow-up fix for https://github.com/gitpod-io/gitpod/pull/18249

## Related Issue(s)

Fixes #

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: components/gitpod-protocol/src/messaging/error.ts [hunk 1]
```diff
@@ -112,7 +112,7 @@ export const ErrorCodes = {
     SNAPSHOT_ERROR: 630 as const,
 
     // 640 Headless logs are not available (yet)
-    HEADLESS_LOG_NOT_YET_AVAILABLE: 640 as const,
+    HEADLESS_LOG_NOT_YET_AVAILABLE: 444 as const,
```
Comment (geropl) [inline components/gitpod-protocol/src/messaging/error.ts:115]: I checked the places where we could filter this from the top of my head:
 - https://github.com/gitpod-io/gitpod/blob/075e14c47f3cadf0e4205fa9a954daf5427b3f70/components/server/src/websocket/websocket-connection-manager.ts#L378-L419 :heavy_check_mark: 
 - https://github.com/gitpod-io/gitpod/blob/b766c9146ebd90da00547eeae718d371c533fc2c/components/gitpod-protocol/src/messaging/error.ts#L30-L34 :heavy_check_mark: 

Anything else we can think of? :thinking:

**Review by geropl — APPROVED** [review]
:heavy_check_mark: to unblock

I think it's worth having an eye on the metrics after rollout in this case, to catch any unexpected filtering on the changed code.

**Comment by easyCZ** [comment]
Dashboard uses a const to check for that particular case, and otherwise there are no relevant occurences of 444. As for filtering, I beleive that as long as it falls in the 4xx category, we're okay.

**Comment by easyCZ** [comment]
@geropl Would you like to take over this PR? If not, I'll close.

**Comment by geropl** [comment]
Please close for now, I'm full atm.

**Pull request closed without merging**